src/freenet/node/PeerManager.java

```diff
@@ -303,7 +346,39 @@ private boolean readPeers(File peersFile, NodeCrypto crypto, OpennetManager open
 		return !someBroken;
 	}
 
-	public boolean addPeer(PeerNode pn) {
+	private void reportDroppedOldPeers(List<String> droppedOldPeers, int droppedOldPeersBuild,
```

Comment:
Can't you stick the user-alert creation method elsewhere? it belongs to freenet/node/useralerts/ imo